source/4_tools/regina/src/regina.cpp

```diff
@@ -1,54 +1,865 @@
+#include "rex_renderer_core/renderer.h"
+#include "rex_renderer_core/resources/mesh.h"
+#include "rex_renderer_core/resources/vertex.h"
+#include "rex_renderer_core/commands/compile_shader_cmd.h"
+#include "rex_renderer_core/commands/create_buffer_cmd.h"
+#include "rex_renderer_core/commands/create_clear_state_cmd.h"
+#include "rex_renderer_core/commands/create_constant_layout_description_cmd.h"
+#include "rex_renderer_core/commands/create_input_layout_cmd.h"
+#include "rex_renderer_core/commands/create_constant_buffer_cmd.h"
+#include "rex_renderer_core/commands/create_raster_state_cmd.h"
+#include "rex_renderer_core/commands/create_pipeline_state_cmd.h"
+#include "rex_renderer_core/commands/link_shader_cmd.h"
+#include "rex_renderer_core/commands/update_commited_resource_cmd.h"
+#include "rex_renderer_core/commands/attach_commited_resource_to_frame_cmd.h"
+#include "rex_renderer_core/rendering/scene.h"
+#include "rex_renderer_core/rendering/scene_renderer.h"
+
 #include "rex_engine/core_application.h"
+#include "rex_engine/entrypoint.h"
+#include "rex_engine/windowinfo.h"
 #include "rex_engine/diagnostics/logging/log_macros.h"
 #include "rex_engine/diagnostics/logging/log_verbosity.h"
-#include "rex_engine/entrypoint.h"
+#include "rex_engine/filesystem/vfs.h"
+#include "rex_engine/frameinfo/frameinfo.h"
+#include "rex_engine/frameinfo/deltatime.h"
+#include "rex_engine/memory/memory_allocation.h"
+#include "rex_engine/primitives/mesh_factory.h"
+#include "rex_engine/primitives/box.h"
+#include "rex_engine/primitives/grid.h"
+#include "rex_engine/primitives/sphere.h"
+#include "rex_engine/primitives/cylinder.h"
+
 #include "rex_std/string.h"
 #include "rex_std_extra/memory/memory_size.h"
+
 #include "rex_windows/gui_application.h"
 #include "rex_windows/platform_creation_params.h"
 
+#include <glm/gtc/matrix_transform.hpp> 
+
 DEFINE_LOG_CATEGORY(LogRegina, rex::LogVerbosity::Log);
 
+#define RENDER_WIREFRAME 0
+#define RENDER_SCENE 1
+
 namespace rex
 {
-  bool initialize()
-  {
-    REX_LOG(LogRegina, "Initializing Regina");
-
-    return true;
-  }
-  void update()
-  {
-    // REX_INFO("updating Regina");
-  }
-  void shutdown()
-  {
-    REX_LOG(LogRegina, "shutting down Regina");
-  }
-
-  ApplicationCreationParams create_regina_app_creation_params(PlatformCreationParams&& platformParams)
-  {
-    ApplicationCreationParams app_params(rsl::move(platformParams));
-
-    app_params.gui_params.window_width  = 1280;
-    app_params.gui_params.window_height = 720;
-    app_params.gui_params.window_title  = "Regina";
-
-    app_params.engine_params.max_memory        = 256_kb;
-    app_params.engine_params.app_init_func     = initialize;
-    app_params.engine_params.app_update_func   = update;
-    app_params.engine_params.app_shutdown_func = shutdown;
-
-    app_params.create_window = true;
-
-    return app_params;
-  }
+    struct ObjectConstants
```

Comment:
These structs will change a lot depending on what we are trying to achieve as they represent the data that is send to the shader. I'd like to keep them here for now until we have a proper solution to build up constant buffers and align them with the shaders being used. I have an idea ( such as the Buffer Layout of the Input Assembler, but I have to figure out how this can be linked within the root signature )